### DISCUSSION TITLE
Port Forwarding: Cannot match any routes

### DISCUSSION THREAD

**Discussion by MitchHamill** [coder/code-server#3069] [Q&A]
I am trying to run an ionic app in my Google cloud server and access it from `/proxy/8100`, when I access the port the page returns only `index.html` and is unable to load any other resources.
The error is: `ERROR Error: Uncaught (in promise): Error: Cannot match any routes. URL Segment: 'proxy/8100'`

Any help is appreciated!
![Screen Shot 2021-04-08 at 3 59 04 pm](https://user-images.githubusercontent.com/57971013/113975828-75d58f80-9883-11eb-9856-1cfd3456e764.png)
![Screen Shot 2021-04-08 at 3 59 08 pm](https://user-images.githubusercontent.com/57971013/113975895-8e45aa00-9883-11eb-8c64-1a888f2ac02a.png)

**Comment by nooobcoder** [comment]
@bpmct any suggestions on this, I too have faced this while building **React App**.
@MitchHamill what I did is, **mapped the port of the container** to the host network using `--net host`.
I have never ever got success with the **proxy/port-number method**, so i exposed few common ports of my container
You can manually expose few common ports too, as

- 3000
- 5000
- like these

or even a range of ports as `-p 1035-1050:1035:1050` **but let me tell you, exposing a range of ports might slow down your container, I have faced this. Docker daemon eats up memory when exposing a range of ports of a container. Proceed with caution**

I got similar errors @MitchHamill when I tried the proxy method,
![image](https://user-images.githubusercontent.com/50350828/114049874-708b3b80-98a9-11eb-85d5-360257565202.png)


It would be better instead if you open specific ports as mentioned above.

**Reply by jsjoeio** [reply]
Hmm... I don't know if the docs would help here but here is a [link](https://github.com/cdr/code-server/blob/main/docs/FAQ.md#why-does-the-code-server-proxy-strip-proxyport-from-the-request-path).

@bpmct might have some insight.

**Reply by bpmct** [reply]
Yeah I don't think these platforms behave well on the routes like `host:port/subfolder/subfolder`, hence these errors. They are built to work on `host:port/`

Some workarounds would be:
- use ngrok to access your ports
- use `code-server --proxy-domain` to proxy the port to a domain you own

However I see you are using --link instead of your own domains. Ideally, we should also give users subdomains for different ports with --link so that they can work on `/`.

**Reply by bpmct** [reply]
Actually @nooobcoder @MitchHamill - Would you be able to try this? https://github.com/cdr/code-server/blob/main/docs/FAQ.md#proxying-to-create-react-app
